src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (typeof compareFunction === 'function') {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = i + 1; j < this.length; j++) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const temp = this[j];
+
+            this[j] = this[i];
+            this[i] = temp;
+          }
+        }
+      }
+
+      return this;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (this[i].toString() > this[j].toString()) {
+          const temp = this[j];
+
+          this[j] = this[i];
+          this[i] = temp;
+        }
+      }
+    }
```

Comment:
When no compareFunction is provided, the elements are converted to strings and sorted according to each character's Unicode code point value. The current implementation only converts to string when comparing, but it should also specify how non-string elements are sorted relative to each other.